src/main/java/com/iota/iri/service/garbagecollector/GarbageCollector.java

```diff
@@ -0,0 +1,323 @@
+package com.iota.iri.service.garbagecollector;
+
+import com.iota.iri.controllers.TipsViewModel;
+import com.iota.iri.service.snapshot.SnapshotManager;
+import com.iota.iri.storage.Tangle;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.*;
+
+/**
+ * This class represents the manager for the cleanup jobs that are issued by the {@link SnapshotManager} in connection
+ * with local snapshots and eventually other parts of the code.
+ *
+ * It plans, manages and executes the cleanup jobs asynchronously in a separate thread so cleaning up does not affect
+ * the performance of the other tasks of the node.
+ */
+public class GarbageCollector {
+    /**
+     * The interval in milliseconds that the garbage collector will check if new cleanup tasks are available.
+     */
+    private static final int GARBAGE_COLLECTOR_RESCAN_INTERVAL = 10000;
+
+    /**
+     * Logger for this class allowing us to dump debug and status messages.
+     */
+    private static final Logger log = LoggerFactory.getLogger(GarbageCollector.class);
+
+    /**
+     * Boolean flag that indicates if the node is being shutdown.
+     */
+    boolean shuttingDown = false;
```

Comment:
they are used by the jobs to check if the garbageCollector is still running - so the alternative would be a protected getter that would allow the jobs to request the status of the manager